android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -121,18 +126,15 @@ protected void onResume() {
     }
 
     private void loadData() {
-        realmRepo.getSessionsByLocation(location)
-                .addChangeListener((sessions, orderedCollectionChangeSet) -> {
-                    this.sessions.clear();
-                    this.sessions.addAll(sessions);
-
-                    setUpcomingSession();
-                    sessionsListAdapter.setCopyOfSessions(sessions);
-                    sessionsListAdapter.notifyDataSetChanged();
-                    if (!Utils.isEmpty(searchText))
-                        sessionsListAdapter.filter(searchText);
-                    handleVisibility();
-                });
+        locationActivityViewModel.getSessionByLocation(location).observe(this,sessionslist -> {
+            sessions=sessionslist;
```

Comment:
Don't assign the instance here. Use ```.clear()``` and then ```addAll``` methods to add the contents of the sessionslist to sessions.
Also follow the standard java naming conventions. It should be ```sessionsList```. capital ```L```.